### PR TITLE
Add annotations for model files, and add auto_annotate_models rake task

### PR DESCRIPTION
**Pull request by philbogle**
Integrate the annotate gem, which automatically annotates model and factory file with comments describing the associated database schema whenever "rake db:migrate" is run in development mode (but not any other mode).

I verified that the db:migrate does not run except in development mode so this will have no impact on our production servers.  From auto_annotate_models.rb:

```
if Rails.env.development?
```

### CODE REVIEW

**Comment by philbogle** [comment]
I am not going to check this in before I come back, but wanted to share what it looks like for comment.

File: dashboard/db/schema.rb [hunk 1]
```diff
@@ -531,7 +531,7 @@
     t.string   "instructions"
     t.datetime "created_at"
     t.datetime "updated_at"
-    t.integer  "phase"
+    t.text     "phase"
```
Comment (laurelfan) [inline dashboard/db/schema.rb]: accidentally checked in?

Comment (philbogle) [inline dashboard/db/schema.rb]: This should go away. This was just an experimental run to see whether we like it what it looks like, before sending it out for actual review I'll sync and rake migrate etc.

**Comment by bcjordan** [comment]
Looks handy!

**Comment by philbogle** [comment]
Josh, PTAL, this is ready to go I think.

**Comment by philbogle** [comment]
(Resending with a proper mention)

@joshlory PTAL, I think this is ready for a look.

File: dashboard/lib/tasks/auto_annotate_models.rake [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      'simple_indexes'          => "false",
+      'model_dir'               => "app/models",
+      'include_version'         => "false",
+      'require'                 => "",
+      'exclude_tests'           => "true",
+      'exclude_fixtures'        => "true",
+      'exclude_factories'       => "false",
+      'exclude_serializers'     => "false",
+      'ignore_model_sub_dir'    => "false",
+      'skip_on_db_migrate'      => "false",
```
Comment (joshlory) [inline dashboard/lib/tasks/auto_annotate_models.rake]: This causes `annotate` to run on `rake db:migrate`?

**Comment by joshlory** [comment]
Cool!  LGTM. :+1:

File: dashboard/lib/tasks/auto_annotate_models.rake (hunk 2 above)
Comment (philbogle) [inline dashboard/lib/tasks/auto_annotate_models.rake]: Correct, annotate runs on "rake db:migrate" in development mode only (which I verified)

**Pull request merged**